gef.py

```diff
@@ -7792,18 +7817,31 @@ def __get_current_block_start_address():
                         if op in extended_registers[exreg]:
                             parameter_set.add(exreg)
 
-        if is_x86_32():
-            nb_argument = len(parameter_set)
-        else:
-            nb_argument = 0
-            for p in parameter_set:
-                nb_argument = max(nb_argument, function_parameters.index(p)+1)
+        nb_argument = -1
+        if function_name.endswith('@plt'):
+            _function_name = function_name.split('@')[0]
+            try:
+                nb_argument = len(libc_args_definitions[_function_name])
+            except:
+                pass
+
+        if nb_argument < 0:
+            if is_x86_32():
+                nb_argument = len(parameter_set)
+            else:
+                for p in parameter_set:
+                    nb_argument = max(nb_argument, function_parameters.index(p)+1)
 
         args = []
         for i in range(nb_argument):
             _key, _value = current_arch.get_ith_parameter(i, in_func=False)
             _value = RIGHT_ARROW.join(DereferenceCommand.dereference_from(_value))
```

Comment:
let's rename this to `_values` and do the `RIGHT_ARROW.join` stuff in the formatting step.

Reply:
ok for _value->_values, but i think dereference_from can return more than two values, therefore handling RIGHT_ARROW in format would be not so clean.
do you agree?

Reply:
I don't mean necessarily in the format line, but in the same group of lines, instead of adding 'formatting-like' stuff up here, let's join the values right before we format it together.

Reply:
ooook